### PR TITLE
fix(VPP):VPP-Displayed-Version-String-Improvements

### PR DESCRIPTION
**Pull request by Lahari-Kamasetty**
Signed-off-by: Lahari-Kamasetty <k.lahari@capgemini.com>

## Prerequisites

- [ ] I have read the [Contributing Guidelines](../blob/main/CONTRIBUTING.md).
- [ ] I have read the [Code of Conduct](../blob/main/CODE_OF_CONDUCT.md).
- [ ] If this is a non-trivial change, I have already opened an accompanying Issue.
- [ ] If applicable, I have included documentation updates alongside my code changes.

## Description
"VPP dispalyed version string improvements"

Introduced new patch - meta-qoriq/recipes-extended/vpp/files/0048-vpp-version-format-change.patch
Modified VPP recipe - meta-qoriq/recipes-extended/vpp/vpp_21.08-lsdk.bb

## Test Plan

Verified on PUMA , PFA output file

vpp-version-format-change

### CODE REVIEW

File: meta-qoriq/recipes-extended/vpp/files/0048-vpp-version-format-change.patch [hunk 1]
```diff
@@ -0,0 +1,21 @@
+diff -ruN a/vpp/app/version.c b/vpp/app/version.c
+--- a/vpp/app/version.c	2022-11-11 12:17:52.665882031 +0530
++++ b/vpp/app/version.c	2022-11-11 12:29:15.631962472 +0530
+@@ -29,7 +29,7 @@
+ 
+ /** The image version string */
+ char *vpe_version_string =
+-  "vpp v" VPP_BUILD_VER
++  "vpp v21.01 " "("VPP_BUILD_VER")"
```
Comment (elludraon) [inline meta-qoriq/recipes-extended/vpp/files/0048-vpp-version-format-change.patch:9]: - Can you remove the redundant quotes?
```
"vpp v21.01 " "("  /* wrong */
"vpp v21.01 ("  /* correct */
```
- Make sure the original `VPP_BUILD_VER` has a "v" prefix as well for consistency, not just the new string.
- I suggest writing `#define VPP_DISPLAY_VER 21.01` and using that define in both places instead.

**Review by elludraon — CHANGES_REQUESTED** [review]
Looks mostly good. Please address a few minor comments. Thanks!

File: meta-qoriq/recipes-extended/vpp/files/0048-vpp-version-format-change.patch [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
++
+ /** \file
+     Display image version information
+ */
+@@ -29,7 +31,7 @@
+ 
+ /** The image version string */
+ char *vpe_version_string =
+-  "vpp v" VPP_BUILD_VER
++  "vpp v" VPP_DISPLAY_VER " ("VPP_BUILD_VER")"
```
Comment (elludraon) [inline meta-qoriq/recipes-extended/vpp/files/0048-vpp-version-format-change.patch:18]: still missing "v" before `VPP_BUILD_VER`

File: meta-qoriq/recipes-extended/vpp/files/0048-vpp-version-format-change.patch [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+   " built by " VPP_BUILD_USER " on " VPP_BUILD_HOST " at " VPP_BUILD_DATE;
+ 
+ /** The name of the compiler */
+@@ -73,8 +75,8 @@
+ 
+   if (verbose)
+     {
++      vlib_cli_output (vm, "%-25s %s (%s)\n", "Version:", "v"VPP_DISPLAY_VER, VPP_BUILD_VER);
+ #define _(a,b,c) vlib_cli_output (vm, "%-25s " b, a ":", c);
+-      _("Version", "%s", "v" VPP_BUILD_VER);
```
Comment (elludraon) [inline meta-qoriq/recipes-extended/vpp/files/0048-vpp-version-format-change.patch:28]: I don't understand why you separated these lines... if you copy/paste the same change you made above, it works here also.

**Review by elludraon — CHANGES_REQUESTED** [review]
See more comments inline. For reference, here is the correct patch:
```diff
From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jeffrey Han <39203126+elludraon@users.noreply.github.com>
Date: Mon Nov 14 13:13:42 2022 -0800
Subject: [PATCH] VPP displayed version string improvements

---
 src/vpp/app/version.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/vpp/app/version.c b/src/vpp/app/version.c
index db13c8639..832ea18a4 100644
--- a/src/vpp/app/version.c
+++ b/src/vpp/app/version.c
@@ -16,6 +16,8 @@
 #include <vppinfra/cpu.h>
 #include <vpp/app/version.h>
 
+#define VPP_DISPLAY_VER "21.01"
+
 /** \file
     Display image version information
 */
@@ -29,7 +31,7 @@
 
 /** The image version string */
 char *vpe_version_string =
-  "vpp v" VPP_BUILD_VER
+  "vpp v" VPP_DISPLAY_VER " (v"VPP_BUILD_VER")"
   " built by " VPP_BUILD_USER " on " VPP_BUILD_HOST " at " VPP_BUILD_DATE;
 
 /** The name of the compiler */
@@ -74,7 +76,7 @@ show_vpe_version_command_fn (vlib_main_t * vm,
   if (verbose)
     {
 #define _(a,b,c) vlib_cli_output (vm, "%-25s " b, a ":", c);
-      _("Version", "%s", "v" VPP_BUILD_VER);
+      _("Version", "%s", "v" VPP_DISPLAY_VER " (v"VPP_BUILD_VER")");
       _("Compiled by", "%s", VPP_BUILD_USER);
       _("Compile host", "%s", VPP_BUILD_HOST);
       _("Compile date", "%s", VPP_BUILD_DATE);
```

And the output:
```
node-4a-57-dd-05-00-01:~# vppctl show version
vpp v21.01 (v21.08.0-LSDK~gdc7be0571) built by  on <host> at 2021-07-19T06:18:04
node-4a-57-dd-05-00-01:~# vppctl show version verbose | head -n1
Version:                  v21.01 (v21.08.0-LSDK~gdc7be0571)
```

File: meta-qoriq/recipes-extended/vpp/files/0048-vpp-version-format-change.patch [hunk 4]
```diff
@@ -0,0 +1,31 @@
+diff -ruN a/vpp/app/version.c b/vpp/app/version.c
+--- a/vpp/app/version.c	2022-11-11 12:17:52.665882031 +0530
++++ b/vpp/app/version.c	2022-11-12 18:52:32.972360041 +0530
```
[the remark this answers is not in the document]
Comment (Lahari-Kamasetty) [reply meta-qoriq/recipes-extended/vpp/files/0048-vpp-version-format-change.patch]: Addressed review comments

File: meta-qoriq/recipes-extended/vpp/files/0048-vpp-version-format-change.patch (hunk 2 above)
Comment (Lahari-Kamasetty) [reply meta-qoriq/recipes-extended/vpp/files/0048-vpp-version-format-change.patch]: Addressed review comments

File: meta-qoriq/recipes-extended/vpp/files/0048-vpp-version-format-change.patch (hunk 3 above)
Comment (Lahari-Kamasetty) [reply meta-qoriq/recipes-extended/vpp/files/0048-vpp-version-format-change.patch]: Addressed review comments

**Review by elludraon — APPROVED** [review]

**Pull request merged**